navapbc/wic-mt-demo-project-mock-api
# Initial mock API foundation

## Ticket

- https://wicmtdp.atlassian.net/browse/WMDP-28
- https://wicmtdp.atlassian.net/browse/WMDP-23

## Changes
> What was added, updated, or removed in this PR.

- Adds initial `.gitignore`
- Creates a docker container and a convenience `docker-compose.yml` for running in local development
- Installs [poetry](https://python-poetry.org) for package management
- Installs [connexion](https://connexion.readthedocs.io/en/stable/) to act as a Flask server that supports OpenAPI
- Creates one `/healthcheck` endpoint

## Context for reviewers
> Testing instructions, background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers. Explain how the changes were verified.

This PR establishes the foundation for the Mock API. It provides initial server and package manager configuration. It also sets up Swagger for local development.

## Testing
> Screenshots, GIF demos, code examples or output to help show the changes working as expected. ProTip: you can drag and drop or paste images into this textbox.

1. Follow the new instructions in the `README.md`
2. Visit `localhost:8080/v1/docs` to access the Swagger UI and test that Swagger works
4. Visit `localhost:8080/v1/healthcheck` to access the healthcheck endpoint and test that the value returned is `true`
